### ISSUE TITLE
`VWRegressor` should have default parameter `convert_labels` set to False

### ISSUE THREAD

**Issue by etiennekintzler** [VowpalWabbit/vowpal_wabbit#3125]
The class `VWRegressor`  has default parameter `convert_labels=False` since it directly inherits from class `VW`.  It should not be the case in a regression setting, as it will convert all continuous data to `{-1, 1}` (from my understanding of the function `tovw`  [in module sklearn_vw](https://github.com/VowpalWabbit/vowpal_wabbit/blob/b06f7678846a4f6ab44bdf6379286b48d0ab7f2d/python/vowpalwabbit/sklearn_vw.py#L1009)).

Here are reproducible python code that illustrates the problem:

```python
from vowpalwabbit.sklearn_vw import VWRegressor
from sklearn.metrics import mean_squared_error
import numpy as np
import pandas as pd

# Data generation process
np.random.seed(0)
n = 200
intercept = 6
coef = np.array([2.4, 1.2])
x = np.random.normal(size=(n, len(coef)))
noise = np.random.normal(size=n, scale=.25)
y = intercept + (x @ coef) + noise

# Models definition
current_default = VWRegressor()
current_default.fit(y=y, X=x)

better_default = VWRegressor(convert_labels=False)
better_default.fit(y=y, X=x)

# Models predictions
y_pred_current_default = current_default.predict(x)
y_pred_better_default = better_default.predict(x)

# Scores
mse_current = mean_squared_error(y_true=y, y_pred=y_pred_current_default)
mse_better = mean_squared_error(y_true=y, y_pred=y_pred_better_default)

# Plot
ax = pd.DataFrame([y, y_pred_current_default, y_pred_better_default]).T.plot(alpha=.6)
ax.legend(["truth", f"current default (mse: {mse_current:.1f})", f"better default (mse: {mse_better:.1f})"])
```

with the plot:

![image](https://user-images.githubusercontent.com/17943873/124645521-d29d1200-de93-11eb-9b69-f04a566de9f1.png)

I suspect this is the issue that @adr-arroyo faces in  https://github.com/VowpalWabbit/vowpal_wabbit/issues/3107 (since he tried regression)

**Comment by jackgerrits** [comment]
Interesting, it seems like surprising behavior for the regressor to be setting the labels to these discrete values. Additionally. it seems from the comment it was not intended to be used in a regression setting from the beginning anyway.

I am in favour of making this change.

**Comment by adr-arroyo** [comment]
Hi, sorry to drop a comment on a closed issue, I have just seen it.

I have tried the following code with the sklearn interface and setting `conver_labels` to `False` (dont mind the data preprocessing). However I am having the same issues as @etiennekintzler for when predicting the values, as you can see in the picture.

```
from vowpalwabbit import sklearn_vw
from vowpalwabbit import pyvw 
from vowpalwabbit.sklearn_vw import tovw

vw_model = pyvw.vw(quiet=False, power_t=0.2)

vw_sklearn_model = sklearn_vw.VWRegressor(convert_labels=False, quiet=False, power_t=0.2)

preds = []
preds_sk = []

fit_every_samples = 1
dif = 0.0
j = 0

for i in range(fit_every_samples, len(data_vw)):

    if j>0:
      preds.append(vw_model.predict(data_vw[j:i][0]))
      preds_sk.append(vw_sklearn_model.predict(x[j:i]))
      
    vw_model.learn(data_vw[j:i][0])
    vw_sklearn_model.fit(x[j:i], y[j:i])
        
    j = i
    i+=fit_every_samples
```

![image](https://user-images.githubusercontent.com/11291505/126451649-082b7272-9ea1-4231-8584-9173381c2e85.png)

![image](https://user-images.githubusercontent.com/11291505/126451682-32bf88e5-18f8-4263-acf1-4308224e7db6.png)

Is it my code that I am not using the sklearn interface as it is supposed?

BR

**Comment by etiennekintzler** [comment]
@adr-arroyo could you post a reproducible example with the data that you used (a few dozen data points should be enough to investigate the problem) ?

**Comment by adr-arroyo** [comment]
Hi,

Just updated my previous post. I included a few more lines with the code to read the data, and also attached it as zip files. The data_joblib.zip file contains the two `x.joblib` and `y.joblib` files. 

Thanks and BR
